src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -947,9 +1296,19 @@ document::value operations::run(entity::map& entity_map,
         auto key = string::to_string(op["arguments"]["client"].get_string().value);
         return fail_point(entity_map, apm_map[key], op_view);
     }
+    if (name == "findOneAndDelete")
+        return find_one_and_delete(
+            entity_map.get_collection(object), get_session(op_view, entity_map), op_view);
+    if (name == "findOneAndReplace")
+        return find_one_and_replace(
+            entity_map.get_collection(object), get_session(op_view, entity_map), op_view);
     if (name == "findOneAndUpdate")
         return find_one_and_update(
             entity_map.get_collection(object), get_session(op_view, entity_map), op_view);
+    if (name == "listCollections") {
+        entity_map.get_database(object).list_collections().begin();
```

Comment:
The implementation of C's `mongoc_database_find_collections_with_opts` is one of the few non-lazy cursors, so this will guarantee that `listCollections` gets run.

Assuming there are no tests that inspect the `listCollections` results, this should be OK. We can add to this later when needed.

Reply:
Great, if we need to examine the `listCollections` results for versioned API, I can just add this in later.